### PR TITLE
Ahmed/vanchor integration tests

### PR DESCRIPTION
**Pull request by AhmedKorim**

### CODE REVIEW

File: packages/sdk-core/src/proving/proving-manager-thread.ts [hunk 1]
```diff
@@ -252,10 +255,21 @@ export class ProvingManagerWrapper {
...
-      pm.setExtDatahash(input.externalDataHash);
-      // leaves insertion
+      const [o1, o2] = pm.setVanchorOutputConfig(...input.outputConfigs) as [JsUtxo, JsUtxo];
       const wasm = await this.wasmBlob;
+      const extData = new wasm.ExtData(
+        input.recipient,
+        input.relayer,
+        input.extAmount,
+        input.fee,
+        o1.commitment,
```
Comment (drewstone) [inline packages/sdk-core/src/proving/proving-manager-thread.ts:265]: This should be an encryption from polkadotjs to start, no? https://polkadot.js.org/docs/util-crypto/examples/encrypt-decrypt/

File: packages/sdk-core/src/proving/proving-manager-thread.ts [hunk 2]
```diff
@@ -252,10 +255,21 @@ export class ProvingManagerWrapper {
...
+        o2.commitment
```
Comment (drewstone) [inline packages/sdk-core/src/proving/proving-manager-thread.ts:266]: Same here, the example in test would be helpful imo. We'd likely want to ensure it follows a similar api to the encryption on the protocol solidity side.

File: tests/integration_tests/substrate/vanchorTests.spec.ts [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+  keyring = {
+    bob,
+    alice,
+    charlie
+  };
+  return keyring;
+}
+
+function generateVAnchorNote(amount: number, chainId: number, outputChainId: number, index?: number) {
+  const noteBuilder = new JsNoteBuilder();
```
Comment (drewstone) [inline tests/integration_tests/substrate/vanchorTests.spec.ts:39]: What's the reason we aren't using sdk-core here again?

File: tests/integration_tests/substrate/vanchorTests.spec.ts [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+    extAmount: extAmount.toString(),
+    fee: fee.toString()
+  };
+  const data = await provingManager.prove('vanchor', setup);
+  const extData = {
+    relayer: address,
+    recipient: address,
+    fee,
+    ext_amount: extAmount,
+    encrypted_output1: data.outputNotes[0].getLeafCommitment(),
```
Comment (drewstone) [inline tests/integration_tests/substrate/vanchorTests.spec.ts:139]: This should be an encrypted value not the commitment.

File: tests/integration_tests/substrate/vanchorTests.spec.ts [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+    encrypted_output2: data.outputNotes[1].getLeafCommitment()
```
Comment (drewstone) [inline tests/integration_tests/substrate/vanchorTests.spec.ts:140]: Same here.

File: tests/integration_tests/substrate/vanchorTests.spec.ts [hunk 6]
```diff
@@ -0,0 +1,365 @@
...
+      extAmount: extAmount.toString(),
+      fee: fee.toString()
+    };
+    const data = await provingManager.prove('vanchor', setup);
+    const extData = {
+      relayer: address,
+      recipient: address,
+      fee,
+      ext_amount: extAmount,
+      encrypted_output1: data.outputNotes[0].getLeafCommitment(),
```
Comment (drewstone) [inline tests/integration_tests/substrate/vanchorTests.spec.ts:266]: Same here.

File: tests/integration_tests/substrate/vanchorTests.spec.ts [hunk 7]
```diff
@@ -0,0 +1,365 @@
...
+      encrypted_output2: data.outputNotes[1].getLeafCommitment()
```
Comment (drewstone) [inline tests/integration_tests/substrate/vanchorTests.spec.ts:267]: Same here.

File: tests/integration_tests/substrate/vanchorTests.spec.ts [hunk 8]
```diff
@@ -0,0 +1,365 @@
...
+      proof: `0x${data.proof}`,
+      publicAmount: data.publicAmount,
+      roots: rootsSet,
+      inputNullifiers: data.inputUtxos.map(input => `0x${input.nullifier}`),
+      outputCommitments: data.outputNotes.map(note => u8aToHex(note.getLeafCommitment())),
+      extDataHash: data.extDataHash
+    };
+    console.log([treeId, vanchorProofData, extData]);
+    try {
+
```
Comment (drewstone) [inline tests/integration_tests/substrate/vanchorTests.spec.ts:280]: Spaces can be removed.

File: tests/integration_tests/substrate/vanchorTests.spec.ts [hunk 9]
```diff
@@ -0,0 +1,365 @@
...
+      extAmount: extAmount.toString(),
+      fee: fee.toString()
+    };
+    const data = await provingManager.prove('vanchor', setup);
+    const extData = {
+      relayer: address,
+      recipient: address,
+      fee,
+      ext_amount: extAmount,
+      encrypted_output1: data.outputNotes[0].getLeafCommitment(),
```
Comment (drewstone) [inline tests/integration_tests/substrate/vanchorTests.spec.ts:343]: Same here.

File: tests/integration_tests/substrate/vanchorTests.spec.ts [hunk 10]
```diff
@@ -0,0 +1,365 @@
...
+      encrypted_output2: data.outputNotes[1].getLeafCommitment()
```
Comment (drewstone) [inline tests/integration_tests/substrate/vanchorTests.spec.ts:344]: Same here.

**Review by drewstone — CHANGES_REQUESTED** [review]
Need to do proper encryptions of the utxo secret data (refer to the [other implementation](https://github.com/webb-tools/protocol-solidity/blob/main/packages/utils/src/Utxo.ts#L86-L94).

File: tests/integration_tests/substrate/vanchorTests.spec.ts (hunk 3 above)
Comment (AhmedKorim) [reply tests/integration_tests/substrate/vanchorTests.spec.ts:39]: I will change it to `sdk-core` for consistency

Comment (drewstone) [reply tests/integration_tests/substrate/vanchorTests.spec.ts:39]: Sounds good, would also be great to change the other tests if we are using wasm-utils directly.

**Review by nepoche — APPROVED** [review]

**Pull request merged**